algosdk/future/atomic_transaction_composer.py

```diff
@@ -0,0 +1,533 @@
+        self.private_key = private_key
+
+    def sign(self, txn_group, indexes):
+        """
+        Sign transactions in a transaction group given the indexes.
+
+        Returns an array of encoded signed transactions. The length of the
+        array will be the same as the length of indexesToSign, and each index i in the array
+        corresponds to the signed transaction from txnGroup[indexesToSign[i]].
+
+        Args:
+            txn_group (list[Transaction]): atomic group of transactions
+            indexes (list[int]): array of indexes in the atomic transaction group that should be signed
+        """
+        stxns = []
+        for i in indexes:
+            stxn = txn_group[i].sign(self.private_key)
+            stxns.append(stxn)
+        return stxns
+
+
+class LogicSigTransactionSigner(TransactionSigner):
+    """
+    Represents a Transaction Signer for a LogicSig that can sign transactions from an
+    atomic transaction group.
+
+    Args:
+        lsig (LogicSigAccount): LogicSig account
+    """
+
+    def __init__(self, lsig) -> None:
+        super().__init__()
+        self.lsig = lsig
+
+    def sign(self, txn_group, indexes):
+        """
+        Sign transactions in a transaction group given the indexes.
+
+        Returns an array of encoded signed transactions. The length of the
+        array will be the same as the length of indexesToSign, and each index i in the array
+        corresponds to the signed transaction from txnGroup[indexesToSign[i]].
+
+        Args:
+            txn_group (list[Transaction]): atomic group of transactions
+            indexes (list[int]): array of indexes in the atomic transaction group that should be signed
+        """
+        stxns = []
+        for i in indexes:
+            stxn = transaction.LogicSigTransaction(txn_group[i], self.lsig)
+            stxns.append(stxn)
+        return stxns
+
+
+class MultisigTransactionSigner(TransactionSigner):
+    """
+    Represents a Transaction Signer for a Multisig that can sign transactions from an
+    atomic transaction group.
+
+    Args:
+        sks (str): private keys of multisig
+    """
+
+    def __init__(self, sks) -> None:
+        super().__init__()
+        self.sks = sks
+
+    def sign(self, txn_group, indexes):
+        """
+        Sign transactions in a transaction group given the indexes.
+
+        Returns an array of encoded signed transactions. The length of the
+        array will be the same as the length of indexesToSign, and each index i in the array
+        corresponds to the signed transaction from txnGroup[indexesToSign[i]].
+
+        Args:
+            txn_group (list[Transaction]): atomic group of transactions
+            indexes (list[int]): array of indexes in the atomic transaction group that should be signed
+        """
+        stxns = []
+        for i in indexes:
+            for sk in self.sks:
+                stxn = txn_group[i].sign(sk)
+            stxns.append(stxn)
```

Comment:
MultisigTransaction returns None from the sign call. 

Believe this should be
```py
      for sk in self.sks:
          txn_group[i].sign(sk)
      stxns.append(txn_group[i])
```

Reply:
Thanks! I will change this in the other branch and merge the changes here soon....